src/app/components/window-terminal/window-terminal.component.ts

```diff
@@ -80,6 +84,28 @@ export class WindowTerminalComponent implements OnInit, OnDestroy {
   
   paste(): void { this.hterm.terminals[this.hterm.currentIndex].output.emit(clipboard.readText()); }
 
+  saveBuffer(): void {
+    let parent: HTMLIFrameElement = <HTMLIFrameElement>document.querySelector('.terminal-instance.active iframe');
+    if (!parent) return;
+    
+    let doc: HTMLDocument = parent.contentWindow.document;
+    let lines: NodeList = doc.querySelectorAll('x-row');
+    let lineBuffer: string[] = [];
+    
+    [].forEach.call(lines, (line :Node) => { 
+      if (line.textContent.length)
+        lineBuffer.push(line.textContent)
+    });
+
+    dialog.showSaveDialog(remote.getCurrentWindow(), (fn) => {
+      if (!fn) return;
+
+      writeFile(fn, lineBuffer.join(EOL), (err) => {
```

Comment:
Sometimes you write a lot of different commands to achive something and then you might want to create a shell script out of that or show somebody the procedure...